tests/test_request.py

```diff
@@ -29,3 +29,28 @@ def test_base64_encoding(time_machine: TimeMachineFixture) -> None:
         url
         == "https://saml.invalid/?SAMLRequest=fZHPSsNAEMZfJew9f%2BnFoSlUixioGtroobd1M7ULyWzc2S16tSC%2Bks/TFzGmKgWxx5n55jffx4xZtk0HU%2B82tMAnj%2ByC57YhhmGQC28JjGTNQLJFBqdgOb2eQxYl0FnjjDKNCKbMaJ02dGGIfYt2iXarFd4t5rnYONcxxLFUHGnaykbXIihmudjv3vavH/vde18yeyyInSSXiyzJkjA5C9NRlY4gSyBNVyIov8%2Bda6o1PZ729nAQMVxVVRmWt8tKBPdouffY86NETMZfCWE4bI8yn8bKn6BiIn2tkRSO4yPQgdrBTb9ZzErTaPUSXBrbSvc/OI3SoaPrcD1IwRN3qPRaY90bjf8yf5vHn5t8Ag%3D%3D"
     )
+
+
+def test_endpoint_url_merge(time_machine: TimeMachineFixture) -> None:
+    time_machine.move_to(
+        datetime.datetime(
+            year=2020,
+            month=9,
+            day=14,
+            hour=14,
+            minute=20,
+            second=11,
+            tzinfo=datetime.timezone.utc,
+        )
+    )
+    url = get_request_redirect_url(
+        saml_endpoint="https://saml.invalid/?idpid=abcdef",
+        expected_audience="audience",
+        acs_url="https://acs.invalid",
+        request_id="テスト",
+        force_reauthentication=False,
+    )
+    assert (
+        url
+        == "https://saml.invalid/?idpid=abcdef&SAMLRequest=fZHPSsNAEMZfJew9f%2BnFoSlUixioGtroobd1M7ULyWzc2S16tSC%2Bks/TFzGmKgWxx5n55jffx4xZtk0HU%2B82tMAnj%2ByC57YhhmGQC28JjGTNQLJFBqdgOb2eQxYl0FnjjDKNCKbMaJ02dGGIfYt2iXarFd4t5rnYONcxxLFUHGnaykbXIihmudjv3vavH/vde18yeyyInSSXiyzJkjA5C9NRlY4gSyBNVyIov8%2Bda6o1PZ729nAQMVxVVRmWt8tKBPdouffY86NETMZfCWE4bI8yn8bKn6BiIn2tkRSO4yPQgdrBTb9ZzErTaPUSXBrbSvc/OI3SoaPrcD1IwRN3qPRaY90bjf8yf5vHn5t8Ag%3D%3D"
+    )
```

Comment:
Does `update_query` guarantee a specific order of query items? I'm a bit worried that this test might be flaky. I think it would be better to parse the URL returned by `get_request_redirect_url` here and check that it has the query items we expect. That means there's also no need to do time travel in this test.

Reply:
Good point! Changed it a bit.